### ISSUE TITLE
Feature Request: Serialised Container Behaviour

### ISSUE THREAD

**Issue by TechJed** [CompositeLight/currentRMS-helper#51]
At the moment if you have a job with just one XLR on it, you would be able to scan all the Video wall onto the job as it has been serialised into containers. Would be great if the plugin could check the job to see if an item in a serialised case is actually on the job before adding it

**Comment by CompositeLight** [comment]
To be clear, I think the scenario you're talking about its:
1. An opportunity has 1 XLR cable on it.
2. There is a container that contains a load of stuff, plus 1 XLR.
3. You scan that container and it'll be accepted because it contains an XLR.
4. All the other (unneeded) contents are added to the job.

I believe I'm correct in saying that vanilla Current won't allow you to scan the containerised XLR (because it's "not available" because it's in a container) so you would have to scan the actual container at step 3. The "extras" get added to a separate group.

What would you like to happen differently?

I guess what you're saying is a warehouse tech scans 6 containers of LED wall, but only meant to scan 5, but because there was also a spare cable on the job somewhere the 6th container became a "valid" scan?

**Comment by TechJed** [comment]
Sorry I didn't explain it that well.
1. An opportunity has only 1 XLR cable on it.
2. There is a container that contains a load of stuff, none of which is on the list .
3. You scan that container and it'll be accepted.
4. All the unneeded contents are added to the job.

We queried this with Current and were told this is the intended behaviour

**Comment by CompositeLight** [comment]
Got it. I think I have run into this before now that you mention it. I don't know why this would be intended behaviour.
Correct behaviour should surely be to block that scan unless Free Scan is enabled. I'm not sure how easy it'll be to do in the extension as I'm not currently pulling container data from anywhere... It'd be an API call per container to get it out, and require constant refreshing. Might be one for the Wish List grave yard I'm afraid.

**Comment by TechJed** [comment]
Yep, thought it might be a little too complex but thought I’d confirm it. Will chase with current again as the behaviour is illogical and probably just laziness. Certainly not what I would be expecting a price hike for
